apps/bootloader.js

```diff
@@ -101,12 +101,28 @@ if (startapp) {
   function drawWidgets() {
     for (var w of WIDGETS) w.draw();
   }
-  var clockApps = require("Storage").list().filter(a=>a[0]=='+').map(app=>{
-    try { return require("Storage").readJSON(app); }
-    catch (e) {}
-  }).filter(app=>app.type=="clock").sort((a, b) => a.sortorder - b.sortorder);
-  if (clockApps && clockApps.length > 0) eval(require("Storage").read(clockApps[0].src));
+  var settings;
+  try {
+    settings = require("Storage").readJSON('@setting');
+  } catch (e) {
+    settings = {}
+  }
+  var clockApp = settings.clock;
+  if (clockApp) {
+      clockApp = require("Storage").read(clockApp)
+  }
+  if (!clockApp) {
+    var clockApps = require("Storage").list().filter(a=>a[0]=='+').map(app=>{
```

Comment:
I'd suggest using `===` as it's safer and a bit faster than `==`

Reply:
Happy to update it - although this was copied directly from the existing code in `firmware.js` so didn't want to rewrite it if particular choices had already been made.

Reply:
>  it's safer and a bit faster than ==

@ra100 have you benchmarked this on Espruino? I'd be tempted to say that it was almost identical, if not marginally slower. In this case I fail to see how it is any safer as we know we're dealing with a string.

Reply:
@gfwilliams point taken, I didn't test it on espruino, just browser :) and you're right, it's marginally slower ~1% or so. And in "safer", I mean, that we can use ===, as we know for sure, we're dealing with strings, so we don't need to rely on == type conversion.